src/components/CromwellModal.js

```diff
@@ -0,0 +1,88 @@
+import _ from 'lodash/fp'
+import { Fragment, useState } from 'react'
+import { div, h } from 'react-hyperscript-helpers'
+import { ButtonPrimary, spinnerOverlay } from 'src/components/common'
+import { withModalDrawer } from 'src/components/ModalDrawer'
+import { tools } from 'src/components/notebook-utils'
+import TitleBar from 'src/components/TitleBar'
+import { Ajax } from 'src/libs/ajax'
+import { withErrorReporting } from 'src/libs/error'
+import Events, { extractWorkspaceDetails } from 'src/libs/events'
+import {
+  computeStyles,
+  getCurrentAppForType
+} from 'src/libs/runtime-utils'
+import * as Utils from 'src/libs/utils'
+
+
+const defaultDataDiskSize = 500 // GB
+const defaultKubernetesRuntimeConfig = { machineType: 'n1-highmem-8', numNodes: 1, autoscalingEnabled: false }
+
+const titleId = 'cromwell-modal-title'
+
+export const CromwellModalBase = Utils.withDisplayName('CromwellModal')(
+  ({
+    onDismiss, onSuccess, apps, appDataDisks, workspace, workspace: { workspace: { namespace, bucketName, name: workspaceName, googleProject } },
```

Comment:
Looks like an unused input?

```suggestion
    onDismiss, onSuccess, apps, workspace, workspace: { workspace: { namespace, bucketName, name: workspaceName, googleProject } },
```

Reply:
It is unused at this point, but I was keeping the input args for this object the same as the input args for the others (for example, GalaxyModalBase). I do imagine we will use appDataDisks at some point.

Would you prefer I delete it anyway?

Reply:
I hear you but for me, it has usually turned out to be the best to not put in/leave things in place with future expectations…probably because the future is difficult to predict 😊

Reply:
Sure, I will remove it. 

Can someone explain to me the `workspace` variables? To me, it appears that workspace is defined as an input twice, but trying to remove the first one (which seems less specific in form) causes a compilation error.

Reply:
FYI, I now understand the workspace variables.